tensorflow-framework/src/main/java/org/tensorflow/framework/losses/CategoricalCrossentropy.java

```diff
@@ -0,0 +1,270 @@
+   *
+   * @param tf the TensorFlow Ops
+   * @param name the name of this loss
+   * @param reduction Type of Reduction to apply to loss.
+   */
+  public CategoricalCrossentropy(Ops tf, String name, Reduction reduction) {
+    this(tf, name, FROM_LOGITS_DEFAULT, LABEL_SMOOTHING_DEFAULT, reduction, DEFAULT_AXIS);
+  }
+
+  /**
+   * Creates a categorical cross entropy Loss using {@link Class#getSimpleName()} as the loss name,
+   * {@link #LABEL_SMOOTHING_DEFAULT} for labelSmoothing, a Loss Reduction of {@link
+   * Loss#REDUCTION_DEFAULT}, and an axis of {@link #DEFAULT_AXIS}
+   *
+   * @param tf the TensorFlow Ops
+   * @param fromLogits Whether to interpret predictions as a tensor of logit values
+   */
+  public CategoricalCrossentropy(Ops tf, boolean fromLogits) {
+    this(tf, null, fromLogits, LABEL_SMOOTHING_DEFAULT, REDUCTION_DEFAULT, DEFAULT_AXIS);
+  }
+
+  /**
+   * Creates a categorical cross entropy Loss using {@link #LABEL_SMOOTHING_DEFAULT} for
+   * labelSmoothing, a Loss Reduction of {@link Loss#REDUCTION_DEFAULT}, and a channel axis of {@link
+   * #DEFAULT_AXIS}
+   *
+   * @param tf the TensorFlow Ops
+   * @param name the name of this loss
+   * @param fromLogits Whether to interpret predictions as a tensor of logit values
+   */
+  public CategoricalCrossentropy(Ops tf, String name, boolean fromLogits) {
+    this(tf, name, fromLogits, LABEL_SMOOTHING_DEFAULT, REDUCTION_DEFAULT, DEFAULT_AXIS);
+  }
+
+  /**
+   * Creates a categorical cross entropy Loss using {@link Class#getSimpleName()} as the loss name,
+   * a Loss Reduction of {@link Loss#REDUCTION_DEFAULT}, and a channel axis of {@link #DEFAULT_AXIS}
+   *
+   * @param tf the TensorFlow Ops
+   * @param fromLogits Whether to interpret predictions as a tensor of logit values
+   * @param labelSmoothing Float in <code>[0, 1]</code>. When <code>&gt; 0</code>, label values are smoothed, meaning the
+   *    confidence on label values are relaxed. e.g. <code>label_smoothing=0.2<code> means that we will use a
+   *    value of </code>0.1<code> for label </code>0<code> and </code>0.9<code> for label </code>1<code>
+   */
+  public CategoricalCrossentropy(Ops tf, boolean fromLogits, float labelSmoothing) {
+    this(tf, null, fromLogits, labelSmoothing, REDUCTION_DEFAULT, DEFAULT_AXIS);
+  }
+
+  /**
+   * Creates a categorical cross entropy Loss using a Loss Reduction of {@link Loss#REDUCTION_DEFAULT},
+   * and a channel axis of {@link #DEFAULT_AXIS}
+   *
+   * @param tf the TensorFlow Ops
+   * @param name the name of this loss
+   * @param fromLogits Whether to interpret predictions as a tensor of logit values
+   * @param labelSmoothing Float in [0, 1]. When 0, no smoothing occurs. When > 0, we compute the
+   *     loss between the predicted labels and a smoothed version of the true labels, where the
+   *     smoothing squeezes the labels towards 0.5. Larger values of label_smoothing correspond to
```

Comment:
This one's still got the doc from BinaryCrossEntropy wrt `label_smoothing`. And it's snake_case.

Reply:
OK